js_modules/dagster-ui/packages/ui-core/src/assets/Stale.tsx

```diff
@@ -127,50 +175,158 @@ function groupedCauses(
   return groupBy(all, (cause) => cause.label);
 }
 
-const StaleCausesPopoverSummary = ({causes}: {causes: LiveDataForNode['staleCauses']}) => (
-  <Box style={{width: '300px'}}>
-    <Box padding={12} border="bottom" style={{fontWeight: 600}}>
-      Changes since last materialization:
-    </Box>
-    <Box style={{maxHeight: '240px', overflowY: 'auto'}} onClick={(e) => e.stopPropagation()}>
-      {causes.map((cause, idx) => (
-        <Box key={idx} border={idx > 0 ? 'top' : null} padding={{vertical: 8, horizontal: 12}}>
-          <Link to={assetDetailsPathForKey(cause.key)}>
-            <CaptionMono>{displayNameForAssetKey(cause.key)}</CaptionMono>
-          </Link>
-          <StaleReason reason={cause.reason} dependency={cause.dependency} />
-        </Box>
-      ))}
+const StaleCausesPopoverSummary = ({
+  assetKey,
+  liveData,
+  include,
+}: {
+  assetKey: AssetKeyInput;
+  liveData?: StaleDataForNode;
+  include: 'all' | 'upstream' | 'self';
+}) => {
+  const grouped = groupedCauses(assetKey, include, liveData);
+  const totalCauses = liveData?.staleCauses?.length;
+
+  if (!totalCauses) {
+    // Should never happen since the parent of this component should check that the asset is stale before rendering the popover
+    return <div />;
+  }
+  return (
+    <Box flex={{direction: 'column'}}>
+      <Box padding={{horizontal: 12, vertical: 8}} border="bottom">
+        <Subtitle2>
+          {numberFormatter.format(totalCauses)} {ifPlural(totalCauses, 'change', 'changes')} since
+          last materialization
+        </Subtitle2>
+      </Box>
+      {Object.entries(grouped).map(([label, causes], idx) => {
+        const isSelf = isEqual(assetKey.path, causes[0]!.key.path);
+        return (
+          <Box key={label}>
+            <Box
+              padding={{horizontal: 12, vertical: 8}}
+              border={idx === 0 ? 'bottom' : 'top-and-bottom'}
+            >
+              <CaptionSubtitle>
+                {getCollapsedHeaderLabel(isSelf, causes[0]!.category, causes.length)}
+              </CaptionSubtitle>
+            </Box>
+            {causes.map((cause, idx) => (
+              <StaleReason
+                key={idx}
+                reason={cause.reason}
+                dependency={cause.dependency}
+                assetKey={assetKey}
+              />
+            ))}
+          </Box>
+        );
+      })}
     </Box>
-  </Box>
-);
+  );
+};
 
 const StaleReason = ({
   reason,
   dependency,
 }: {
+  assetKey: AssetKeyInput;
   reason: string;
   dependency: AssetNodeKeyFragment | null;
 }) => {
-  if (!dependency) {
-    return <Caption>{` ${reason}`}</Caption>;
-  }
+  const content = useMemo(() => {
```

Comment:
Not super important really but I don't know that this is particularly worth a memo, we do way worse in each render cycle 🙈  Though maybe we should be doing this!

Reply:
I separated the changed reason to a new component so I can remove this now! 😄